lib/ansible/modules/network/enos/enos_config.py

```diff
@@ -0,0 +1,291 @@
+        any changes without affecting how the set of commands are matched
+        against the system.
+    required: false
+    default: null
+  after:
+    description:
+      - The ordered set of commands to append to the end of the command
+        stack if a change needs to be made.  Just like with I(before) this
+        allows the playbook designer to append a set of commands to be
+        executed after the command set.
+    required: false
+    default: null
+  match:
+    description:
+      - Instructs the module on the way to perform the matching of
+        the set of commands against the current device config.  If
+        match is set to I(line), commands are matched line by line.  If
+        match is set to I(strict), command lines are matched with respect
+        to position.  If match is set to I(exact), command lines
+        must be an equal match.  Finally, if match is set to I(none), the
+        module will not attempt to compare the source configuration with
+        the running configuration on the remote device.
+    required: false
+    default: line
+    choices: ['line', 'strict', 'exact', 'none']
+  replace:
+    description:
+      - Instructs the module on the way to perform the configuration
+        on the device.  If the replace argument is set to I(line) then
+        the modified lines are pushed to the device in configuration
+        mode.  If the replace argument is set to I(block) then the entire
+        command block is pushed to the device in configuration mode if any
+        line is not correct.
+    required: false
+    default: line
+    choices: ['line', 'block', 'config']
+  config:
+    description:
+      - The module, by default, will connect to the remote device and
+        retrieve the current running-config to use as a base for comparing
+        against the contents of source.  There are times when it is not
+        desirable to have the task get the current running-config for
+        every task in a playbook.  The I(config) argument allows the
+        implementer to pass in the configuration to use as the base
+        config for comparison.
+    required: false
+    default: null
+  backup:
+    description:
+      - This argument will cause the module to create a full backup of
+        the current C(running-config) from the remote device before any
+        changes are made.  The backup file is written to the C(backup)
+        folder in the playbook root directory.  If the directory does not
+        exist, it is created.
+    required: false
+    default: no
+    choices: ['yes', 'no']
+    version_added: "2.5"
+  comment:
+    description:
+      - Allows a commit description to be specified to be included
+        when the configuration is committed.  If the configuration is
+        not changed or committed, this argument is ignored.
+    required: false
+    default: 'configured by enos_config'
+    version_added: "2.5"
+  admin:
```

Comment:
1) Is this the same as EOS & IOS's privilege mode?

2) Would a password ever be needed

3) In Ansible 2.5 we are moving to using the standard Ansible 'become` system to enter enable mode, though that requires `connection: network_cli`, which I don't believe you are currently using.

Reply:
For ENOS, to get into privilege mode, by default , u dont need a password. Soon after login using admin/admin u are in exec mode. To go to Privilege Mode u just have to type in enable. I have already made in the sample / test roles in my Github with connection: network_cli. But I am facing an issue in my module_util file enos.py that in load_config method its coming as exec mode (> prompt, not # prompt ) I am at the moment testing it with this code 
def load_config(module, config):
    try:
        conn = get_connection(module)
        conn.get('enable')
        conn.edit_config(config)
    except ConnectionError as exc:
        module.fail_json(msg=to_text(exc))
May be this will change once I put become=true and become_method = enable. 
But as of now its not working for me